integration_test/Dialect/Pipeline/nonstallable/nonstallable_helper.py

```diff
@@ -0,0 +1,53 @@
+import cocotb
+from cocotb.triggers import Timer
+import cocotb.clock
+
```

Comment:
I would appreciate it if you could add a randomized test, (random start time/duration of stall signal, pipeline not necessarily totally full when stall starts).

Reply:
Added a somewhat exhaustive test for fill cycles vs. stall cycles. I'm still not 100% confident in the implementation, but the traces look right to me - I wouldn't be surprised if we're going to see some more bugs pop up, but let's cross that bridge once we push some code through it.